heudiconv/bids.py

```diff
@@ -103,8 +103,10 @@ def maybe_na(val: Any) -> str:
         return "n/a"
 
 
-def treat_age(age: str|float) -> str:
+def treat_age(age: str | float | None) -> str | None:
     """Age might encounter 'Y' suffix or be a float"""
+    if age is None:
+        return None  # might be converted to N/A by maybe_na
     agestr = str(age)
     if agestr.endswith("M"):
```

Comment:
If `age` is limited to `str | float`, then `if agestr.endswith("M")` will never be true.  Why is this branch here?

Reply:
Disregard this; I somehow read `str` as `int`.